src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -472,4 +473,8 @@ public static Map<String, String> convertStringToObjectMapToStringToStringMap(Ma
             return stringToStringMap;
         }
     }
+
+    public static Boolean checkIfInputsContainsKey(Map<String, Object> inputs, String enumKey){
```

Comment:
This method is good but the naming implies we're just checking whether the key exists.  How about `parseBooleanIfExists` or something similar?   

Also as a more generic util method it may not be "inputs` so just a more generic map variable name; and the key may be a string constant, not necessarily an enum so just `key` probably works there.

This needs a Javadoc added explaining what the method does.

Reply:
Method name was made to be more generic and the 'enumKey' parameter is changed to just 'key', javaDoc too was added

Reply:
Hey @martinpkr! Thanks for raising the issue. That was really fast since the time I created the issue. In #718 I have also added inputs for [Float](https://github.com/opensearch-project/flow-framework/pull/718/files#diff-598529bfc64aa4f2a77a460b76f73ba2ac76bae8df01f1eb92a937223396438dR94-R96) as well. This method can be make generic to support Float, Boolean for now and later we can keep on extending it.

Reply:
Made the method use generics, now it should be easy to expand it, also made some more tests, tell me if there should be more changes to this :wrench:

Reply:
Since https://github.com/opensearch-project/flow-framework/pull/718 is merged. Can you incorporate the changes in ReIndexWorkflowStep file as well?

Reply:
All done!

Reply:
Ah! Looks like your PR got changes from my PR. Can you resolve the conflicts and push only your changes?

Reply:
Resolved conflicts in PR.